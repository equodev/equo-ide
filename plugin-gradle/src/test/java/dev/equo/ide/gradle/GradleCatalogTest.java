/*******************************************************************************
 * Copyright (c) 2022-2025 EquoTech, Inc. and others.
 *
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     EquoTech, Inc. - initial API and implementation
 *******************************************************************************/
package dev.equo.ide.gradle;

import static org.junit.jupiter.api.condition.JRE.JAVA_17;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import java.io.IOException;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.condition.EnabledForJreRange;
import org.junit.jupiter.api.extension.ExtendWith;

@ExtendWith({SnapshotExtension.class})
public class GradleCatalogTest extends GradleHarness {
	private void test(String content, Expect expect) throws IOException {
		setFile("build.gradle")
				.toLines(
						"plugins { id 'dev.equo.ide' }",
						"equoIde {",
						content,
						"  addFilter 'platform-neutral', { platformNone() }",
						"}");
		run("equoList", "--request", "--stacktrace")
				.snapshotBetween("Task :equoList", "BUILD SUCCESSFUL", expect);
	}

	@Test
	@EnabledForJreRange(min = JAVA_17)
	public void simple(Expect expect) throws IOException {
		test("jdt('4.27')", expect.scenario("jdt"));
		test("platform('4.27')\ngradleBuildship()", expect.scenario("gradleBuildship"));
	}

	@Test
	public void versionOverride(Expect expect) throws IOException {
		test("jdt('4.25')", expect.scenario("jdt-spec"));
		test("platform()\njdt('4.25')", expect.scenario("platform-neutral-jdt-spec"));
		test("platform('4.25')\njdt()", expect.scenario("platform-spec"));
		test("platform('4.25')\njdt('4.25')", expect.scenario("both-spec"));
	}

	@Test
	public void wrongOrder(Expect expect) throws IOException {
		setFile("build.gradle")
				.toLines(
						"plugins { id 'dev.equo.ide' }",
						"equoIde {",
						"  jdt()",
						"  platform()",
						"  addFilter 'platform-neutral', { platformNone() }",
						"}");
		runAndFail("equoList", "--request", "--stacktrace")
				.snapshotBetween(
						"A problem occurred evaluating root project 'under-test'.", "* Try:", expect);
	}
}
