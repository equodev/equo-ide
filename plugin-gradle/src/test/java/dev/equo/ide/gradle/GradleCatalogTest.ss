╔═ simple/gradleBuildship ═╗
+-------------+-----------------------------------------------------------------------------------------+
| kind        | value                                                                                   |
+-------------+-----------------------------------------------------------------------------------------+
| p2repo      | https://download.eclipse.org/buildship/updates/e431/releases/3.x/3.1.10.v20240802-1211/ |
| p2repo      | https://download.eclipse.org/eclipse/updates/4.27/                                      |
| install     | org.eclipse.buildship.feature.group                                                     |
| install     | org.eclipse.platform.ide.categoryIU                                                     |
| install     | org.eclipse.releng.java.languages.categoryIU                                            |
| filter      | platform-neutral                                                                        |
|   osgi.arch | dont-include-platform-specific-artifacts                                                |
|   osgi.os   | dont-include-platform-specific-artifacts                                                |
|   osgi.ws   | dont-include-platform-specific-artifacts                                                |
+-------------+-----------------------------------------------------------------------------------------+
╔═ simple/jdt ═╗
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.27/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | platform-neutral                                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+
╔═ versionOverride/both-spec ═╗
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.25/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | platform-neutral                                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+
╔═ versionOverride/jdt-spec ═╗
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.25/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | platform-neutral                                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+
╔═ versionOverride/platform-neutral-jdt-spec ═╗
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.25/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | platform-neutral                                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+
╔═ versionOverride/platform-spec ═╗
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.25/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | platform-neutral                                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+
╔═ wrongOrder ═╗
> platform was already added as a transitive dependency of jdt.
  You can fix this by moving the <platform> block above the <jdt> block.
╔═ [end of file] ═╗
