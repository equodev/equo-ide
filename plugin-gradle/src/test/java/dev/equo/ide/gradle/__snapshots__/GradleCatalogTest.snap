dev.equo.ide.gradle.GradleCatalogTest.simple[gradleBuildship]=[
+-------------+-----------------------------------------------------------------------------------------+
| kind        | value                                                                                   |
+-------------+-----------------------------------------------------------------------------------------+
| p2repo      | https://download.eclipse.org/buildship/updates/e431/releases/3.x/3.1.10.v20240802-1211/ |
| p2repo      | https://download.eclipse.org/eclipse/updates/4.27/                                      |
| install     | org.eclipse.buildship.feature.group                                                     |
| install     | org.eclipse.platform.ide.categoryIU                                                     |
| install     | org.eclipse.releng.java.languages.categoryIU                                            |
| filter      | platform-neutral                                                                        |
|   osgi.arch | dont-include-platform-specific-artifacts                                                |
|   osgi.os   | dont-include-platform-specific-artifacts                                                |
|   osgi.ws   | dont-include-platform-specific-artifacts                                                |
+-------------+-----------------------------------------------------------------------------------------+
]


dev.equo.ide.gradle.GradleCatalogTest.simple[jdt]=[
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.27/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | platform-neutral                                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+
]


dev.equo.ide.gradle.GradleCatalogTest.versionOverride[both-spec]=[
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.25/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | platform-neutral                                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+
]


dev.equo.ide.gradle.GradleCatalogTest.versionOverride[jdt-spec]=[
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.25/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | platform-neutral                                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+
]


dev.equo.ide.gradle.GradleCatalogTest.versionOverride[platform-neutral-jdt-spec]=[
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.25/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | platform-neutral                                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+
]


dev.equo.ide.gradle.GradleCatalogTest.versionOverride[platform-spec]=[
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.25/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | platform-neutral                                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+
]


dev.equo.ide.gradle.GradleCatalogTest.wrongOrder=[
> platform was already added as a transitive dependency of jdt.
  You can fix this by moving the <platform> block above the <jdt> block.
]